.pre-commit-config.yaml

```diff
@@ -49,13 +49,20 @@ repos:
       - id: golangci-lint
         args: [--new-from-patch=/tmp/diff.patch]
 
-    # Rules are in .markdownlint-cli2.yaml file
-    # See https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md for rule descriptions
+  # Rules are in .markdownlint-cli2.yaml file
+  # See https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md for rule descriptions
   - repo: https://github.com/DavidAnson/markdownlint-cli2
     rev: v0.8.1
     hooks:
-    - id: markdownlint-cli2-fix
+      - id: markdownlint-cli2-fix
+
+  # Rules are in .yamllint.yaml file
+  # See https://yamllint.readthedocs.io/en/stable/rules.html# for rule descriptions
+  - repo: https://github.com/adrienverge/yamllint.git
+    rev: v1.29.0
+    hooks:
```

Comment:
is it possible to cover formatting of lists?
for example, here we have:
```yaml
   hooks:
      - id: yamllint
```

but in Deployment manifests:
```yaml
      volumes:
      - name: nginx
        emptyDir: { }
        emptyDir: {}
```
note the indentation of `-`.

I vote for the latter style.

Reply:
Yeah I think so - the indentation rule is marked as "consistent" (so all indentation in a particular file is the same, but it doesn't cover across files). I'll see what I can come up with!

Reply:
Ok I checked, and yes this can be configured using `indent-sequences` (see https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.indentation). I realise now that I actually changed this to `consistent` from the default of `true`. I changed it to `false` and this does require all lists be unindented. However, after I changed a few files, I don't know if unindenting all the configuration files is the right thing to do? All the CI workflow files and all the dotconfig etc files use this pattern, and for the most part, this style was copied from the docs for these tools, and seems to be the generally accepted pattern. I also think it's easier to read for these files. Meanwhile, unindented lists is the accepted pattern for k8s manifests, so it makes sense to keep that consistent there.

Reply:
is it possible to configure it per directory?
Otherwise, a Kubernetes manifests, inconsistent with the accepted pattern for k8s manifests, can be merged

Reply:
Not really. An option would be to create a second indentation rule (which is actually breaking yaml syntax, but it will run) with the `indent-sequences` set to `false` and list all the files and folders we want to ignore, e.g.

```
  indentation:
    spaces: consistent
    indent-sequences: false
    check-multi-line-strings: true
    ignore: |
      deploy/manifests/nginx-gateway.yaml
      deploy/manifests/njs-modules.yaml
      .github/
      *.yml
      .pre-commit-config.yaml
      deploy/helm-chart/Chart.yaml
```
This will catch any indented lists in any yaml files not listed there (I can't find a way to do it in a '!' not files pattern, nothing else I've tried works), but I don't really like this approach - any further config files would have to be added here, and I personally get an annoying red line in my IDE because of the duplicate key.

Additionally, our deployment manifests are now generated, and all current k8s manifests are following the k8s pattern, so we only really have any new example k8s manifests files that could potentially break that pattern.